### PR TITLE
Support of directives on fields and types during introspection.

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: src/StrawberryShake/CodeGeneration/test/CodeGeneration.Tests/Utilities/__snapshots__/QueryDocumentRewriterTests.GetReturnTypeName.snap [hunk 1]
```diff
@@ -2,12 +2,6 @@
   hero(episode: NEW_HOPE) {
     __typename
     ... Characters
-    ... on Droid {
-      id
-    }
-    ... on Human {
-      id
-    }
```
Comment (PascalSenn) [inline src/StrawberryShake/CodeGeneration/test/CodeGeneration.Tests/Utilities/__snapshots__/QueryDocumentRewriterTests.GetReturnTypeName.snap:10]: Is this related to this PR?

Comment (michaelstaib) [reply src/StrawberryShake/CodeGeneration/test/CodeGeneration.Tests/Utilities/__snapshots__/QueryDocumentRewriterTests.GetReturnTypeName.snap:10]: not really.

Comment (michaelstaib) [reply src/StrawberryShake/CodeGeneration/test/CodeGeneration.Tests/Utilities/__snapshots__/QueryDocumentRewriterTests.GetReturnTypeName.snap:10]: Must be from a merge ... will have a look at this tomorrow... well spotted 👍

Comment (michaelstaib) [reply src/StrawberryShake/CodeGeneration/test/CodeGeneration.Tests/Utilities/__snapshots__/QueryDocumentRewriterTests.GetReturnTypeName.snap:10]: OK... this is from the merge the query is ok
